src/main/java/org/assertj/core/internal/Files.java

```diff
@@ -287,6 +288,30 @@ public void assertCanWrite(AssertionInfo info, File actual) {
     throw failures.failure(info, shouldBeWritable(actual));
   }
 
+  /**
+   * Asserts that the given {@code File} is empty (i.e. size is equal to zero bytes).
+   * @throws AssertionError if the given {@code File} is {@code null}.
+   * @throws AssertionError if the given {@code File} does not exist.
+   * @throws AssertionError if the given {@code File} is not empty.
+   */
+  public void assertIsEmptyFile(AssertionInfo info, File actual) {
+    assertIsFile(info, actual);
+    if (actual.length() == 0) return;
+    throw failures.failure(info, shouldBeEmpty());
+  }
+
+  /**
+   * Asserts that the given {@code File} is not empty (i.e. size is greater than zero bytes).
+   * @throws AssertionError if the given {@code File} is {@code null}.
+   * @throws AssertionError if the given {@code File} does not exist.
+   * @throws AssertionError if the given {@code File} is empty.
+   */
+  public void assertIsNotEmptyFile(AssertionInfo info, File actual) {
+    assertIsFile(info, actual);
+    if (actual.length() > 0) return;
+    throw failures.failure(info, shouldNotBeEmpty());
```

Comment:
introduce `shouldNotBeEmpty*(File)` with message `"%nExpecting file <%s> not to be empty"` (add the corresponding test in `ShouldNotBeEmpty_create_Test`.